core/src/main/java/org/arquillian/smart/testing/impl/ConfiguredSmartTestingImpl.java

```diff
@@ -0,0 +1,61 @@
+package org.arquillian.smart.testing.impl;
+
+import java.io.File;
+import java.util.Set;
+import org.arquillian.smart.testing.Configuration;
+import org.arquillian.smart.testing.api.ConfiguredSmartTesting;
+import org.arquillian.smart.testing.api.TestStrategyApplier;
+import org.arquillian.smart.testing.api.TestVerifier;
+import org.arquillian.smart.testing.spi.JavaSPILoader;
+
+public class ConfiguredSmartTestingImpl implements ConfiguredSmartTesting {
```

Comment:
I'm just wondering do we really need interfaces and classes implemeting them? Would we ever have 2 implementations for one interface considering DSL usage?

Reply:
Well, probably we don't need it - we could get rid of the `ConfiguredSmartTesting` interface as well as the `TestExecutionPlannerLoader` interface. In case of `TestStrategyApplier` we have two different implementations. 
I used this approach (using interfaces) as IMHO it is a good practice in the case of API...

Reply:
What does it improve in this given case?